libraries/Microsoft.Bot.Builder.Skills/SkillHandler.cs

```diff
@@ -44,7 +44,7 @@ public class SkillHandler : ChannelServiceHandler
         public SkillHandler(
             BotAdapter adapter,
             IBot bot,
-            ISkillConversationIdFactory conversationIdFactory, 
+            ISkillConversationIdFactory conversationIdFactory,
```

Comment:
I'm still not up-to-date on this interface; but here we are saying that developers should always use an `ISkillConversationIdFactory`.

Since we don't include any implementation of our own, this (to me) feels like an odd half-measure, I could just return the same conversationId. It's not clear that I need to create a new conversationId after looking at the [interface](https://github.com/microsoft/botbuilder-dotnet/blob/e2ea0d6e5a32ad0fc298471639b4ef3c3012ece3/libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/Skills/ISkillConversationIdFactory.cs#L9).

Some developers may not be worried about obfuscating conversationIds or don't want to transform their data back and forth because they own the caller and the skills. 

___

I'm not against this interface, but I'm not entirely sure what it is addressing, and having it be required without an implementation or explanation doesn't sit well with me.

Reply:
we can discuss more in person but the purpose if this interface is to provide a way of restoring the original ServerUrl parameter in the activity so it can be rerouted back to the channel. 
Due to how the protocol works the implementation is obscure and I will add a default implementation based on ConversationState that developers can use but can't bake that in because we can't assume ConversationState will be always there (unless we make it a hard requirement for skills). 
I also have planned a SkillHttpClient that derives from BotFrameworkHttpClient that takes this interface so the class will be used in outgoing and incoming requests from the skill (using SkillHttpClient and SkillHandler). But I can't put this in yet because conversationId impacts the OAuthPrompt flow and I need to fix that first before adding the new classes.